python/ray/experimental/state.py

```diff
@@ -386,3 +386,118 @@ def task_profiles(self):
+        worker = workers[info["worker_id"]]
+        if i > 0:
+          f.write(",\n")
+        if parent_info:
+          parent_worker = workers[parent_info["worker_id"]]
+          parent_times = self.get_times(parent_info)
+          f.write(json.dumps({
+              "cat": "submit_task",
+              "pid": "Node " + str(parent_worker["node_ip_address"]),
+              "tid": parent_worker["worker_index"],
+              "ts": micros(min(parent_times)),
+              "ph": "s",
+              "name": "SubmitTask",
+              "args": {},
+              "id": str(i),
+          }))
+          f.write(",\n")
+          f.write(json.dumps({
+              "cat": "submit_task",
+              "pid": "Node " + str(worker["node_ip_address"]),
+              "tid": worker["worker_index"],
+              "ts": micros(min(times)),
+              "ph": "f",
+              "name": "SubmitTask",
+              "args": {},
+              "id": str(i),
+          }))
+          f.write(",\n")
+        f.write(json.dumps({
+            "name": info["function_name"],
+            "cat": "ray_task",
+            "ph": "X",
+            "ts": micros(min(times)),
+            "dur": micros(max(times)) - micros(min(times)),
+            "pid": "Node " + str(worker["node_ip_address"]),
+            "tid": worker["worker_index"],
+            "args": info
+        }))
+      f.write("]")
+    task_info
+
+  def get_times(self, data):
+    all_times = []
+    all_times.append(data["acquire_lock_start"])
+    all_times.append(data["acquire_lock_end"])
+    all_times.append(data["get_arguments_start"])
+    all_times.append(data["get_arguments_end"])
+    all_times.append(data["execute_start"])
+    all_times.append(data["execute_end"])
+    all_times.append(data["store_outputs_start"])
+    all_times.append(data["store_outputs_end"])
+    return all_times
+
+  def actor_classes(self):
+    actor_info = dict()
+    actors = self.redis_client.keys("ActorClass*")
+    actor_classes = dict()
+    for actor in actors:
+      actor_key_str = actor[len("Actor:"):]
+      actor_info["ActorClass:{}".format(binary_to_hex(actor))] = (
+          self.redis_client.hgetall(actor))
+      actor_key = actor_info["ActorClass:{}".format(binary_to_hex(actor))]
+      actor_classes[binary_to_hex(actor)] = {
+          "driver_id": binary_to_hex(actor_key[b"driver_id"]),
+          "class": actor_key[b"class"],
+          "class_name": actor_key[b"class_name"].decode("ascii"),
+          "module": actor_key[b"module"].decode("ascii"),
+          "actor_method_names": (actor_key[b"actor_method_names"]
+                                 .decode("ascii")),
+      }
+    return actor_classes
+
+  def workers(self):
+    workers = self.redis_client.keys("Worker*")
+    worker_info = dict()
+    workers_data = dict()
+    i = 0
+    for worker in workers:
+      worker_key = worker[len("Workers:"):]
+      worker_info["Workers:{}".format(binary_to_hex(worker_key))] = (
```

Comment:
Please double check every place where we call `binary_to_hex` in this PR. This should only be called on IDs. Calling it on the worker ID here is fine, but it shouldn't be called on `node_ip_address` or `plasma_manager_socket`, etc.